File: problems-extend-adapt/3201-3300/3274_matching-square-colors/solution.cpp
class Solution {
  public:
    bool squaresShareColor(string coordinate1, string coordinate2) {
        // A square's color follows the parity of column index plus row
        // number; character-code offsets are even, so raw codes keep it.
        int p1 = (coordinate1[0] + coordinate1[1]) % 2;
        int p2 = (coordinate2[0] + coordinate2[1]) % 2;
        return p1 == p2;
    }
};

File: problems-extend-adapt/3201-3300/3274_matching-square-colors/solution.rs
impl Solution {
    pub fn squares_share_color(coordinate1: String, coordinate2: String) -> bool {
        // A square's color follows the parity of column index plus row
        // number; character-code offsets are even, so raw codes keep it.
        let b1 = coordinate1.as_bytes();
        let b2 = coordinate2.as_bytes();
        let p1 = (b1[0] + b1[1]) % 2;
        let p2 = (b2[0] + b2[1]) % 2;
        p1 == p2
    }
}
